src/pages/dashboard.js

```diff
@@ -5,3 +5,13 @@ import Dashboard from "../components/dashboard";
 export default function DashboardPage() {
     return <Dashboard />;
 }
+
+DashboardPage.getInitialProps = async ({ Component, ctx }) => {
+    let pageProps = {};
+
+    if (Component.getInitialProps) {
+        pageProps = await Component.getInitialProps(ctx);
+    }
+
+    return { pageProps, namespacesRequired: ["dashboard"] };
+};
```

Comment:
I removed the `getInitialProps` from `_apps.js` and moved it here cos I was seeing warning in the console
`You have not declared a namespacesRequired array on your page-level component: DashboardPage. This will cause all namespaces to be sent down to the client, possibly negatively impacting the performance of your app.`

Also see - 

https://github.com/isaachinman/next-i18next#4-declaring-namespace-dependencies

Reply:
So if I'm understanding correctly, then each page in the `pages` directory will need to have this same thing but with whatever message files that page will need?

Reply:
Thats correct.

Reply:
That way it won't download translation files for the whole app. Instead only files related to that page or downloaded.

Reply:
OK, so if during development my component needs a message from a file my page hasn't used before, I have to add it to the `useTranslation` hook in my component and also remember to add it to the page's `getInitialProps`? The `getInitialProps` part might be a little annoying / easy to forget. I was hoping we could have what you had before with one global `getInitialProps` that uses `fs` to import all the message files automatically and never have to worry about updating it manually, but I still think the convenience we get at the component level is far, far better than what we have with `react-intl`.  I'm still for moving forward with this change 👍

Reply:
No. You don't update the `getInitialProps` during development. I just double checked. Updating just the `useTranslation` with namespace should work. As I said, this is a performance optimization so that the page won't download all the translation file the app.

Reply:
Just make sure you do update the `getInitialProps` when you put a PR :) / changing something for production deployment.

Reply:
Got it - that's not that bad then!  We'll just all need to be mindful during PR review to catch that too.  Hopefully we get an error or something to catch it quickly otherwise.